### PR TITLE
Add lesson flow with lecture answers and questions, perform migration and apis generation

### LINKED ISSUE #72: oll-263-design-of-lesson-structure-and-user-flow-in-the-course

### PR DESCRIPTION
**Pull request by Toumi-Elyes**
### Summary
🗄️📝🎓

This pull request implements a new course structure with lectures, sections, and lessons, and adds new features and models for answers, user progress, and lesson status. It also updates the `docker-compose.yml` file, the database schema, and some service methods and tests to improve consistency and readability. It removes the unused `Chapter` table and related files and classes.

> _Sing, O Muse, of the mighty pull request, that reshaped the course content and the user progress_
> _How the skillful coder, `AnswerService`, created and updated many answers with ease_
> _And how the wise `CourseService` returned the course ids, instead of the string messages of old_
> _And how the brave `LessonService` joined the users to the lessons, and used the sections, not the chapters, as they were told_

### Walkthrough
*  Add, modify, and remove files, models, and fields to support new features and refactor course structure ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L1-R1),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L23),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L37-R44),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L77),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-911c2f26afe0a3ab23d133c256fa92140a6ebf6d921aca608fe699467bd5ffeeR1-R91),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L39-L45),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215R57),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L74),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215R75-R94),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L95-R124),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L135-R168),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L152-R177),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L187-R215),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-62024469dd0472709983f6ff32c14657a854c706ad975690a2f8451b62b0aff7),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-58aece3fa4b13acbdfb15008c61a15e94c3f9eba251ebde23d9ffcf817f8ba43),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5b4d04969a34a5b060ba512944f74da95e68074bcde6c735e06c54d682095b3f),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-1c38230112128105acd79b0fa4e3b9722cd9294f80a5025801bd45ea5593d335),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L55-R55),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L112-R112),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL66-R66),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL156-R156),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL311-R311),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8L43-R44),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8L107-R108),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L11-R11),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L25-R25),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L46-R46),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL23-R24),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL77-R78),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL131))
  * Add new tables, columns, and constraints to the database schema using SQL commands in `migration.sql` ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-911c2f26afe0a3ab23d133c256fa92140a6ebf6d921aca608fe699467bd5ffeeR1-R91))
  * Remove the `PermissionChapter` enum and the `Chapter` model from the `schema.prisma` file, as they are no longer needed ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L39-L45),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L95-R124))
  * Add the `UsertoLesson` field to the `User` model, the `LessonStatus` enum to the `schema.prisma` file, and the `UsertoLesson` model to store user progress and status for each lesson ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215R57),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215R75-R94))
  * Remove the `permission_chapter` field from the `UsertoCourse` model, as it is no longer needed ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L74))
  * Modify the `Section` model to have a one-to-many relation with the `Lesson` model, instead of the `Chapter` model ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L95-R124))
  * Modify the `Lesson` model to have one-to-many relations with the `Question` and `Lecture` models, instead of the `Question` model only ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L95-R124))
  * Add the `Lecture` model to store the lecture content for each lesson ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L95-R124))
  * Remove the `data` field from the `Question` model, and add the `difficulty`, `picture`, and `points` fields to store the difficulty level, picture id, and points awarded for each question ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L135-R168))
  * Remove the `point` and `value` fields from the `Answer` model, and add the `data` and `picture` fields to store the answer text and picture id for each answer ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L152-R177))
  * Modify the `Picture` model to have one-to-many relations with the `Question` and `Answer` models, instead of the `Course` model only ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-d0444251deaf3de66fe0760ea9e5bca05582fd0b3744818b6bb68be97d2a5215L187-R215))
  * Delete the files for the `ChapterModule`, `ChapterService`, `ChapterController`, and `ChapterSpec`, as they are no longer needed ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-62024469dd0472709983f6ff32c14657a854c706ad975690a2f8451b62b0aff7),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-58aece3fa4b13acbdfb15008c61a15e94c3f9eba251ebde23d9ffcf817f8ba43),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5b4d04969a34a5b060ba512944f74da95e68074bcde6c735e06c54d682095b3f),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-1c38230112128105acd79b0fa4e3b9722cd9294f80a5025801bd45ea5593d335))
  * Change the return values of the `postCourse`, `deleteCourse`, and `updateCourse` methods in the `CourseService` class from string messages to course ids, and update the corresponding test cases in the `CourseSpec` file ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L55-R55),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L112-R112),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL66-R66),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL156-R156),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL311-R311))
  * Replace the `chapter_id` field by the `section_id` field in the `LessonModel`, `CreateLessonModel`, and `UpdateLessonModel` classes in the `lesson.dto.ts` file, and in the `postLesson`, `getLesson`, and `updateLesson` methods in the `LessonService` and `LessonController` classes, to reflect the new course structure that uses sections instead of chapters ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8L43-R44),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8L107-R108),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L11-R11),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L25-R25),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L46-R46),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL23-R24),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL77-R78))
  * Remove the `data` field from the `getQuestion` method in the `LessonService` class, as it is no longer needed ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL131))
  * Change the `version` value in the `docker-compose.yml` file from single quotes to double quotes, and remove some empty lines, for consistency or style reasons ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L1-R1),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L23),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L77))
  * Change the `test` value in the `healthcheck` section of the `postgres` service in the `docker-compose.yml` file from a single line to a multi-line array, for readability or formatting reasons ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L37-R44))
* Add, modify, and remove files, models, and fields to simplify the responses and make them more consistent with other methods that return ids ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L55-R55),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L112-R112),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL66-R66),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL156-R156),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL311-R311),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5ca534280a656e7c46ef6f5540d09f676f732a6afe220c780d373416af4d9050L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-8729704df41fe71477d8d33e3cd688f3509bbc3db7c3463fc41cb9bcd3c5420cL69-R69),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL33-R34),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL53-R54),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL103-R104))
  * Change the return values of the `postCourse`, `deleteCourse`, and `updateCourse` methods in the `CourseService` class from string messages to course ids, and update the corresponding test cases in the `CourseSpec` file ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L55-R55),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a0fc3c9c0b5dcde246033a12b90f5ccbca8be0799d7f21fce4d3d92d4f269594L112-R112),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL66-R66),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL156-R156),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-4056522b0490544912a0bf818a35a26f8037ade95487bc3aac472bdc16e60d1eL311-R311))
  * Change the return value of the `postDiscussion` method in the `DiscussionService` class from a string message to the discussion id, and update the corresponding test case in the `DiscussionSpec` file ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5ca534280a656e7c46ef6f5540d09f676f732a6afe220c780d373416af4d9050L35-R35),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-8729704df41fe71477d8d33e3cd688f3509bbc3db7c3463fc41cb9bcd3c5420cL69-R69))
  * Change the return values of the `postLesson`, `deleteLesson`, and `updateLesson` methods in the `LessonService` class from string messages to lesson ids ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL33-R34),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL53-R54),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL103-R104))
* Add files, models, and fields to create a new endpoint for joining a lesson, which takes the lesson id as a parameter and the user id as a body, and returns the user lesson id ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8R21),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8R138-R161),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L2-R2),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65R58-R63),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cR10),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL15-R16),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cR139-R161))
  * Add the import of the `IsNumber` validator to the `lesson.dto.ts` file, to use the validator for validating the `points` field in the `QuestionModel` class ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65L2-R2))
  * Add the `JoinLessonModel` class to the `lesson.dto.ts` file, to define the model for joining a lesson, which contains the user id as a field ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-3c963d2fa5c9c9ef6e9e2db667e4d00deb6fb81cd8846d468d843e121c6b5b65R58-R63))
  * Add the import of the `JoinLessonModel` to the `lesson.service.ts` file, to use the model for validating the input for joining a lesson ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cR10))
  * Add the import of the `UsertoLesson` model to the `lesson.service.ts` file, to use the model for accessing the database table for user lessons ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cL15-R16))
  * Add the `joinLesson` method to the `LessonService` class in the `lesson.service.ts` file, to create a new service for joining a lesson, which takes the lesson id and the user id as inputs, and creates a new user lesson record in the database, and returns the user lesson id ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-073d5b9f83c83ae5e42b822fb8df91aaab9221309fe2198a4e79d0e8fec4dd3cR139-R161))
  * Add the import of the `JoinLessonModel` to the `lesson.controller.ts` file, to use the model for validating the request body for joining a lesson ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8R21))
  * Add the `joinLesson` method to the `LessonController` class in the `lesson.controller.ts` file, to create a new endpoint for joining a lesson, which takes the lesson id as a parameter and the user id as a body, and returns the user lesson id ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-21c03bd3078604712564828a4f384d4dbadcbc1eed37495086bd6e939c2389c8R138-R161))
* Add files and classes to create a controller, a service, and a module for handling requests related to answers, such as creating, deleting, updating, and getting answers, and to ensure the quality and functionality of the answer service, by testing its methods with different scenarios and expectations ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a1994ef74e2ed70f58cc8c6d8bae0ceb3a930afc1ba20ff06d2cfb912a28765fR1-R121),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L2-R4),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L11-R39),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L42-R59),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5b3d14620033ccf497a7950a7b91a7ebddf790d36695e366ca6f6efd056f0400R1-R10),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-669f4e3a4a45bc0684bfe52d1604b16b27a0c9783d99d33bd89397d5d6670101R1-R120),[link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a2f28df838433b8ba9a145938d9ab09d85a23e6833a2e0b66af72bcfd956cb04R1-R297))
  * Rename the file `chapter.dto.ts` to `answer.dto.ts`, and add the import of the `IsNumber` validator, to reuse the existing file for defining the data transfer objects for answers, instead of chapters ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L2-R4))
  * Rename the `ChapterModel` class to `AnswerModel` in the `answer.dto.ts` file, and replace the `chapterId` field by the `questionId` field, to define the model for answers, instead of chapters, as answers belong to questions, not chapters ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L11-R39))
  * Rename the `CreateChapterModel` class to `CreateAnswerModel` in the `answer.dto.ts` file, and replace the `chapter_id` field by the `questionId` field, to define the model for creating answers, instead of chapters, as answers belong to questions, not chapters ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-e20fea911e5b48863cd7e5249155d6bd470a109ff93ce2bbcbb5436699ca44d6L42-R59))
  * Add a new file `answer.controller.ts`, containing the `AnswerController` class, to create a controller for handling requests related to answers ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a1994ef74e2ed70f58cc8c6d8bae0ceb3a930afc1ba20ff06d2cfb912a28765fR1-R121))
  * Add a new file `answer.service.ts`, containing the `AnswerService` class, to create a service for handling the business logic and database operations related to answers ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-669f4e3a4a45bc0684bfe52d1604b16b27a0c9783d99d33bd89397d5d6670101R1-R120))
  * Add a new file `answer.module.ts`, containing the `AnswerModule` class, to create a module for grouping the providers and controllers related to answers ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-5b3d14620033ccf497a7950a7b91a7ebddf790d36695e366ca6f6efd056f0400R1-R10))
  * Add a new file `answer.spec.ts`, containing the unit tests for the `AnswerService` class, to ensure the quality and functionality of the answer service ([link](https://github.com/Ollamy/Ollamy/pull/73/files?diff=unified&w=0#diff-a2f28df838433b8ba9a145938d9ab09d85a23e6833a2e0b66af72bcfd956cb04R1-R297))

### CODE REVIEW

File: packages/service-backend/src/answer/answer.service.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    } catch (error) {
+      Logger.error(error);
+      if (error instanceof Prisma.PrismaClientKnownRequestError) {
+        throw new ConflictException('Answer already removed !');
+      }
+      throw new ConflictException('Answer not created !');
+    }
+  }
+
+  async getAnswer(AnswerId: string): Promise<AnswerModel> {
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/answer/answer.service.ts]: Why dont re use the `IdAnswerModel`

File: packages/service-backend/src/answer/answer.service.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          : undefined,
+      } as AnswerModel;
+    } catch (error) {
+      Logger.error(error);
+      throw new ConflictException('Answer not found !');
+    }
+  }
+
+  async updateAnswer(
+    AnswerId: string,
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/answer/answer.service.ts]: Why dont re use the `IdAnswerModel``

File: packages/service-backend/src/lesson/lesson.dto.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class UpdateLessonModel {
   @ApiProperty()
   @IsUUID()
   @IsOptional()
-  chapterId: string;
+  sectionId: string;
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/lesson/lesson.dto.ts]: add `?` if the variable is optional

File: packages/service-backend/src/question/question.dto.ts [hunk 4]
```diff
@@ -86,20 +108,26 @@ export class UpdateQuestionModel {
   @ApiProperty()
   @IsString()
   @IsOptional()
-  data: string;
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/question/question.dto.ts:89]: add `?` if the variable is optional

File: packages/service-backend/src/question/question.dto.ts [hunk 5]
```diff
@@ -86,20 +108,26 @@ export class UpdateQuestionModel {
...
+  @IsEnum(QuestionDifficulty)
+  @IsOptional()
+  difficulty?: QuestionDifficulty;
 
   @ApiProperty()
-  @IsNumber()
+  @IsUUID()
   @IsOptional()
-  point: number;
+  trustAnswerId: string;
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/question/question.dto.ts]: add `?` if the variable is optional

File: packages/service-backend/src/user/user.dto.ts [hunk 6]
```diff
@@ -63,19 +57,10 @@ export class LoginUserModel {
...
   password: string;
 }
 
 export class UpdateUserModel {
-  @IsUUID()
-  id: string;
-
   @ApiProperty()
   @IsOptional()
   firstname: string;
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/user/user.dto.ts]: add `?` if the variable is optional

File: packages/service-backend/src/user/user.dto.ts [hunk 7]
```diff
@@ -92,12 +77,6 @@ export class UpdateUserModel {
   @ApiProperty()
   @IsString()
   @IsOptional()
-  @IsStrongPassword({
-    minLength: 8,
-    minNumbers: 2,
-    minUppercase: 1,
-    minSymbols: 1,
-  })
   password: string;
```
Comment (ThisisYoYoDev) [inline packages/service-backend/src/user/user.dto.ts]: add `?` if the variable is optional

**Review by ThisisYoYoDev — CHANGES_REQUESTED** [review]

File: packages/service-backend/src/question/question.dto.ts (hunk 4 above)
Comment (Toumi-Elyes) [reply packages/service-backend/src/question/question.dto.ts:89]: you are talking about a removed line

**Review by ThisisYoYoDev — APPROVED** [review]

**Pull request merged**